modules/EvseManager/IECStateMachine.cpp

```diff
@@ -505,17 +505,26 @@ void IECStateMachine::connector_force_unlock() {
         cp = cp_state;
     }
 
+    if (not relais_on) {
+        // Unconditionally try to unlock, as `is_locked` might not always reflect the physical state of the lock.
+        // This can occur for example in case of a failed unlock due to a hardware issue.
+        signal_unlock();
+        is_locked = false;
+    }
+
     if (cp == RawCPState::B or cp == RawCPState::C) {
         force_unlocked = true;
         check_connector_lock();
     }
 }
 
 void IECStateMachine::check_connector_lock() {
-    if (should_be_locked and not force_unlocked and not is_locked) {
+    bool should_be_locked_considering_relais_and_force = relais_on or (should_be_locked and not force_unlocked);
+
+    if (not is_locked and should_be_locked_considering_relais_and_force) {
         signal_lock();
         is_locked = true;
-    } else if ((not should_be_locked or force_unlocked) and is_locked and not relais_on) {
+    } else if (is_locked and not should_be_locked_considering_relais_and_force) {
```

Comment:
This will always unlock if relais are off, i.e. always in state B. This is however not what we want, as in public spaces it should really only be unlocked in state A if not forced.